include/fmt/format.h

```diff
@@ -2752,7 +2752,7 @@ template <typename T, FMT_ENABLE_IF(std::is_floating_point<T>::value&&
                                         has_isfinite<T>::value)>
 FMT_CONSTEXPR20 bool isfinite(T value) {
   constexpr T inf = T(std::numeric_limits<double>::infinity());
-  if (is_constant_evaluated())
+  if (is_constant_evaluated(true))
```

Comment:
One drawback is that it switches to a slightly worse implementation on older systems: https://www.godbolt.org/z/WWzjEsrxj. But I guess we can optimize it later.

Reply:
I don't see an alternative other than testing `FMT_CONSTEXPR20` in the body to replace  `is_constant_evaluated` :(

Reply:
We can use ``__builtin_is_constant_evaluated`` instead of ``std::is_constant_evaluated``.

Reply:
Or we can improve 

https://github.com/fmtlib/fmt/blob/89860eb9013a345608c8144b1aad5f12b0682d7e/include/fmt/core.h#L108-L110

condition by adding a stdlib version check.

@hchataing, which version of libc++ or libstdc++ you are using?

Reply:
@hchataing check PR #3754 please

Reply:
Neither macro is defined by the compiler... but maybe I am checking wrong (`clang++ -std=c++20 -E -dM foo.cc`)